compositor_render/src/renderer/scene.rs

```diff
@@ -0,0 +1,194 @@
+    TransformParams,
+};
+use log::error;
+
+use crate::{
+    registry::GetError,
+    transformations::{shader::Shader, web_renderer::WebRenderer},
+};
+
+use super::{
+    texture::{OutputTexture, Texture},
+    RenderCtx,
+};
+
+pub struct InputNode {}
+
+pub enum TransformNode {
+    Shader {
+        params: HashMap<String, ShaderParams>,
+        shader: Arc<Shader>,
+    },
+    WebRenderer {
+        renderer: Arc<WebRenderer>,
+    },
+    Nop,
+}
+
+impl TransformNode {
+    fn new(ctx: &RenderCtx, spec: &TransformParams) -> Result<Self, GetError> {
+        match spec {
+            TransformParams::WebRenderer { renderer_id } => Ok(TransformNode::WebRenderer {
+                renderer: ctx.web_renderers.get(renderer_id)?,
+            }),
+            TransformParams::Shader {
+                shader_id,
+                shader_params,
+            } => Ok(TransformNode::Shader {
+                params: shader_params.clone(),
+                shader: ctx.shader_transforms.get(shader_id)?,
+            }),
+        }
+    }
+    pub fn render(&self, ctx: &RenderCtx, sources: &[(&NodeId, &Texture)], target: &Texture) {
+        match self {
+            TransformNode::Shader {
+                params: _,
+                shader: _,
+            } => {
+                // shader.render(sources, target, other_args);
+                error!("mocked shader render")
+            }
+            TransformNode::WebRenderer { renderer } => {
+                if let Err(err) = renderer.render(ctx, sources, target) {
+                    error!("Render operation failed {err}");
+                }
+            }
+            TransformNode::Nop => (),
+        }
+    }
+}
+
+pub struct Node {
+    pub node_id: NodeId,
+    pub output: Texture,
+    pub resolution: Resolution,
+    pub inputs: Vec<Arc<Node>>,
+    pub transform: TransformNode,
+}
+
+impl Node {
+    pub fn new(
+        ctx: &RenderCtx,
+        spec: &TransformNodeSpec,
+        inputs: Vec<Arc<Node>>,
+    ) -> Result<Self, GetError> {
+        let node = TransformNode::new(ctx, &spec.transform_params)?;
+        let output = Texture::new_rgba(ctx.wgpu_ctx, &spec.resolution);
+        Ok(Self {
+            node_id: spec.node_id.clone(),
+            transform: node,
+            resolution: spec.resolution,
+            inputs,
+            output,
+        })
+    }
+
+    pub fn new_input(ctx: &RenderCtx, spec: &InputSpec) -> Result<Self, GetError> {
+        let output = Texture::new_rgba(ctx.wgpu_ctx, &spec.resolution);
+
+        Ok(Self {
+            node_id: spec.input_id.0.clone(),
+            transform: TransformNode::Nop,
+            resolution: spec.resolution,
+            inputs: vec![],
+            output,
+        })
+    }
+}
+
+pub struct Scene {
+    pub outputs: HashMap<OutputId, (Arc<Node>, OutputTexture)>,
+    pub inputs: HashMap<InputId, (Arc<Node>, [Texture; 3])>,
+}
```

Comment:
Why we can't keep those textures in `Nodes`? Also, I think that we should create the `YuvTexture` struct for this array`[Texture; 3]` for readability.

Reply:
> Why we can't keep those textures in Nodes? 

- I think it's cleaner (single responsibility and all that)
- Those values represent yuv textures, so only some nodes will have them. RGBA textures are in every node, so they are part of the Node struct.

> I think that we should create the YuvTexture

Alerady done in follow up PR where I'm working on yuv rgba conversion